libs/blocks/global-navigation/blocks/navDropdown/dropdown.js

```diff
@@ -0,0 +1,270 @@
+import { getAnalyticsValue, toFragment, decorateCta } from '../../utilities/utilities.js';
+import { localizeLink } from '../../../../utils/utils.js';
+
+const decorateHeadline = (elem) => {
+  if (!(elem instanceof HTMLElement)) return null;
+
+  // TODO: proper handling of aria-expanded across devices
+  const headline = toFragment`<div
+    class="feds-popup-headline"
+    role="heading"
+    aria-level="2"
+    aria-haspopup="true"
+    aria-expanded="false">
+    ${elem.textContent.trim()}
+  </div>`;
+
+  // TODO: move proper logic to accessibility,
+  // this is just for demo functionality purposes
+  headline.addEventListener('click', (e) => {
+    e.preventDefault();
+
+    const openPopup = document.querySelector('.feds-popup-headline[aria-expanded = "true"]');
+
+    if (openPopup && openPopup !== headline) {
+      openPopup.setAttribute('aria-expanded', 'false');
+    }
+
+    const currentState = headline.getAttribute('aria-expanded');
+    headline.setAttribute('aria-expanded', currentState === 'false');
+  });
+
+  // Since heading is turned into a div, it can be safely removed
+  elem.remove();
+
+  return headline;
+};
+
+const decorateLinkGroup = (elem, index) => {
+  if (!(elem instanceof HTMLElement) || !elem.querySelector('a')) return null;
+
+  // TODO: allow links with image and no label
+  const image = elem.querySelector('picture');
+  const link = elem.querySelector('a');
+  const description = elem.querySelector('p:nth-child(2)');
+
+  const imageElem = image ? toFragment`<div class="feds-navLink-image">${image}</div>` : '';
+  const descriptionElem = description ? toFragment`<div class="feds-navLink-description">${description.textContent}</div>` : '';
+  const contentElem = link ? toFragment`<div class="feds-navLink-content">
+      <div class="feds-navLink-title">${link.textContent}</div>
+      ${descriptionElem}
+    </div>` : '';
+  const linkGroup = toFragment`<a 
+    href="${localizeLink(link.href)}"
+    class="feds-navLink"
+    daa-ll="${getAnalyticsValue(link.textContent, index)}">
+      ${imageElem}
+      ${contentElem}
+    </a>`;
+
+  return linkGroup;
+};
+
+// Decorate a list of simple links
+const decorateLinkList = ({ list, className = 'feds-navLink', includeCta = true } = {}) => {
+  list.querySelectorAll('a').forEach((link, index) => {
```

Comment:
Unfortunately the `decorateButtons` method adds a lot of classes and markup that aren't useful for our use-cases, so we'd rather keep it contained to what we actually need instead of re-using a method and then manipulating the result yet again.